src/libostree/ostree-repo-checkout.c

```diff
@@ -436,10 +437,25 @@ checkout_file_hardlink (OstreeRepo                          *self,
             if (!glnx_fstatat (destination_dfd, destination_name, &dest_stbuf,
                                AT_SYMLINK_NOFOLLOW, error))
               return FALSE;
-            const gboolean is_identical =
+            gboolean is_identical =
               (src_stbuf.st_dev == dest_stbuf.st_dev &&
                src_stbuf.st_ino == dest_stbuf.st_ino);
+            if (!is_identical && (src_stbuf.st_size == dest_stbuf.st_size))
+              {
+                /* resort to a checksum check */
```

Comment:
Can you add a little bit more info here; could just be circular with a link to this PR, which I tend to do since it saves people time in `git annotate` if the code moves around.

Like:
```
  /* It might happen that the file came from another repo, as is currently the
   * case in rpm-ostree package layering.  We resort to doing a full checksum.
   * https://github.com/ostreedev/ostree/pull/1258
   */
```